File: domain/entities/global_deck.go
package entities

import "time"

type GlobalDeck struct {
	ID          int       `json:"id"`
	Title       string    `json:"title"`
	Description *string   `json:"description"`
	AuthorID    int       `json:"author_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	IsPublic    bool      `json:"is_public"`
	Tags        []string  `json:"tags"`
	Version     int       `json:"version"`
}

File: Client/lib/domain/entities/GlobalDeck.dart
import 'dart:core';

class searchResultDeck{
  final int ID;
  final String title;
  final String? description;
  final int authorID;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool isPublic;
  final List<String>? tags;
  final int version;
  searchResultDeck({required this.ID,
  required this.title,
  this.description,
    required this.authorID,
    required this.createdAt,
    required this.updatedAt,
    required this.isPublic,
    this.tags,
    required this.version
  });
  searchResultDeck copyInfoWith({
    String? title,
    String? description,
    List<String>? tags,
  }) {
    return searchResultDeck(
      ID:  ID,
      title: title ?? this.title,
      description: description ?? this.description,
      authorID: authorID,
      createdAt: createdAt,
      updatedAt: updatedAt,
      isPublic: isPublic,
      tags: tags ?? this.tags,
      version: version
    );
  }
}